Stub TestCoverage API

rschnekenbu opened the pull request:
#### What it does
Stub the new API introduced with vscode 1.88 about TestCoverage. This was previously a proposed API.

Fixes https://github.com/eclipse-theia/theia/issues/13549

Contributed on behalf of STMicroelectronics

#### How to test

I did not have a testing plugin yet, as usual plugins used for testing are not implementing the coverage API. For example, Playwright does not yet implement code coverage as seen here: https://github.com/microsoft/playwright/issues/30298. 

#### Follow-ups

Implement TestCoverage API. This will be tracked as a task of https://github.com/eclipse-theia/theia/issues/12010.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

tsmaeder commented on packages/plugin-ext/src/plugin/types-impl.ts:
```diff
@@ -3330,6 +3330,77 @@ export class TestMessage implements theia.TestMessage {
+    constructor(public name: string, public executed: number | boolean, public location: Position | Range) { }
+}
+@es5ClassCompat
+export class TestCoverageCount implements theia.TestCoverageCount {
```
`theia.TestCoverageCount` is not an interface. Why the "implements"?

rschnekenbu replied:
I removed this declaration, that indeed makes no sense.

tsmaeder commented on packages/plugin-ext/src/plugin/types-impl.ts:
```diff
@@ -3330,6 +3330,77 @@ export class TestMessage implements theia.TestMessage {
+}
+
+export namespace StatementCoverage {
+    export function is(coverage: unknown): coverage is StatementCoverage {
```
Wouldn't `instanceof` work just as well, since this is a class?

rschnekenbu replied:
Good catch, I removed this namespace addition in favor of a instanceof.

tsmaeder commented:
@rschnekenbu we should at least have a test plugin that exercises the stubbed code: extensions should at least not crash when they call the API.

tsmaeder reviewed (commented):
The `FileCoverage` API is weird in that you can assign to the public fields, but changes to those fields are not handled. Seems to be that same in VS Code, though. Seems undercooked for a GA API on their side, IMO.

rschnekenbu commented:
I addressed the review comments in a following patch. 
Here is a testing extension that adds coverage support for the test runs and also provides the optional method TestRunProfile.loadCoverageDetail for our TestRunProfile.
src: [test-extension-sample-0.1.0-src.zip](https://github.com/eclipse-theia/theia/files/15092078/test-extension-sample-0.1.0-src.zip)
extension: 
[test-extension-sample-0.1.0.zip](https://github.com/eclipse-theia/theia/files/15092084/test-extension-sample-0.1.0.zip)

To run the extension:
1. Start theia browser example on a workspace and install the provided extension
2. Use the command to create a test run profile 
3. Then in the test run view, run the tests. They should run as usual, even if a coverage is added. No errors should be logged or displayed. 

Here is the behavior on vscode 1.88:
![vscode-13549](https://github.com/eclipse-theia/theia/assets/3964263/88a2d999-b741-4813-bf23-8f3a7b5e6173)

And on theia browser example:
![theia-13549](https://github.com/eclipse-theia/theia/assets/3964263/5f7f6ed0-4dc0-4ac9-b392-f5a49f51347d)
